src/shell-integration/bash/ghostty.bash

```diff
@@ -95,6 +95,113 @@ if [[ "$GHOSTTY_SHELL_FEATURES" == *"sudo"* && -n "$TERMINFO" ]]; then
   }
 fi
 
+# SSH Integration
+if [[ "$GHOSTTY_SHELL_FEATURES" == *ssh-* ]]; then
+  ssh() {
+    builtin local ssh_term ssh_opts
+    ssh_term=""
+    ssh_opts=()
+
+    # Configure environment variables for remote session
+    if [[ "$GHOSTTY_SHELL_FEATURES" == *ssh-env* ]]; then
+      ssh_opts+=(-o "SetEnv COLORTERM=truecolor")
+      ssh_opts+=(-o "SendEnv TERM_PROGRAM TERM_PROGRAM_VERSION")
+    fi
+
+    # Install terminfo on remote host if needed
+    if [[ "$GHOSTTY_SHELL_FEATURES" == *ssh-terminfo* ]]; then
+      builtin local ssh_config ssh_user ssh_hostname
+      ssh_config=$(builtin command ssh -G "$@" 2>/dev/null)
+
+      while IFS=' ' read -r ssh_key ssh_value; do
+        case "$ssh_key" in
+          user) ssh_user="$ssh_value" ;;
+          hostname) ssh_hostname="$ssh_value" ;;
+        esac
+        [[ -n "$ssh_user" && -n "$ssh_hostname" ]] && break
+      done <<< "$ssh_config"
```

Comment:
I think we can drop the `ssh_config` intermediate variable.

```suggestion
      done < <(builtin command ssh -G "$@" 2>/dev/null)
```

(This also avoids the subshell.)